crates/noirc_evaluator/src/ssa_refactor/acir_gen/acir_ir/acir_variable.rs

```diff
@@ -31,6 +31,9 @@ pub(crate) struct AcirContext {
     /// then the `acir_ir` will be populated to assert this
     /// addition.
     acir_ir: GeneratedAcir,
+
+    /// Maps an `AcirVar` to its known bit size.
+    variables_to_bit_sizes: HashMap<AcirVar, u32>,
```

Comment:
Previous SSA was doing type().bits, we store a map of variables to known bit sizes. A variable can be constrained multiple times, so we can open an issue to delay all range constraints being added until we call `finalize` and then just take the smallest range.

Note, that if along with equality sets mentioned [here](https://github.com/noir-lang/noir/issues/1431) we can optimize for the case where a variable has been range constrained and then it has been asserted to be equal to another variable. The latter variable can then skip its range constraint.

Reply:
Putting this as #1439